src/main/scala/com/johnsnowlabs/nlp/annotators/TokenizerModel.scala

```diff
@@ -349,21 +351,17 @@ class TokenizerModel(override val uid: String)
                       .length > 1)
                     val applyChars = isSet(splitChars) && splitCharsExists.exists(target.contains)
                     if (target.nonEmpty && (applyPattern || applyChars)) {
-                      try {
-                        val strs =
-                          if (applyPattern) target.split($(splitPattern))
-                          else target.split($(splitChars).mkString("|"))
-                        strs.map(str =>
-                          try {
-                            IndexedToken(
-                              str,
-                              text.start + candidate.start + curPos,
-                              text.start + candidate.start + curPos + str.length - 1)
-                          } finally {
-                            curPos += str.length + 1
-                          })
-                      } finally {
-                        curPos -= 1
```

Comment:
I don't know why initially there were these `try-finally` blocks without catching any exceptions or handling any resources. I removed them as we would want to be specific with the exceptions (and easier maintenance if a bug would occur here).

Reply:
@maziyarpanahi 

It was part of the issue, due to the way the indexes were calculated (always adding and subtracting 1). 

However, I think that having a log line there would be pretty useful. I will add it in a new commit!

Reply:
In case it is needed I openen the PR here:
https://github.com/JohnSnowLabs/spark-nlp/pull/13291